pkg/autodiscovery/providers/env_test.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package providers
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	logsConfig "github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnvCollectEmpty(t *testing.T) {
+	envProvider := NewEnvProvider()
+
+	integrationConfigs, err := envProvider.Collect()
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(integrationConfigs))
+}
+
+func TestEnvCollectValid(t *testing.T) {
+	config.Datadog.Set("logs_config.custom_configs", "[{\"type\":\"tcp\",\"port\":1234,\"service\":\"fooService\",\"source\":\"barSource\",\"tags\":[\"foo:bar\",\"baz\"]}]")
+	envProvider := NewEnvProvider()
+
+	integrationConfigs, err := envProvider.Collect()
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(integrationConfigs))
+
+	var logConfig []logsConfig.LogsConfig
+	err = json.Unmarshal(integrationConfigs[0].LogsConfig, &logConfig)
```

Comment:
you should be using `logsConfig.ParseJSON`

Reply:
also I am not sure we should test logs-config from here, again, autodiscovery should be agnostic from logs-specific logics

Reply:
Agreed, this looks like tests for `logsConfig.ParseJSON`

Reply:
I think we should at least test that this is not config when the env var is not set.